src/utils/api.ts

```diff
@@ -6,7 +6,7 @@ import '../../types';
  * @param {object} response
  * @returns {promise}
  */
-export function handleResponseType(response: Response): Promise<[boolean, Blob | string] | void> {
+export function handleResponseType(response: Response): Promise<[boolean, any]> {
```

Comment:
Just wondering why do we remove types here.

Reply:
Because it also returns json:
https://github.com/verdaccio/ui/blob/f84571251912aae71463c44822c84ca70f8b8c40/src/utils/api.ts#L16
So in this case it returns the tuple `[boolean, any]` since `.json()` returns `any`.

So to be technically correct the return type should be `Promise<[boolean, Blob | string | any]>`, but if you have a union type like this and at any part there is `any`, it will collapse to `any`.

Now you could write it like `Promise<[boolean, Blob | string | any]>` but then 
https://github.com/verdaccio/ui/blob/f84571251912aae71463c44822c84ca70f8b8c40/src/utils/api.ts#L59
errors, since the signature for the `request` method looks like this:
https://github.com/verdaccio/ui/blob/f84571251912aae71463c44822c84ca70f8b8c40/src/utils/api.ts#L33
You would then need a typecast at `resolve(response[1] as any);` which I tried to avoid or go down a rabbit hole of type errors/potential rewrite of the ajax logic.

As I said the ajax functionality is a little bit cumbersome to type as it is. The main cause being that the return type is determined dynamically at runtime (in the `handleResponseType` function), thus there is no natural way for TypeScript to tell the actual return type without explicit casts for the current code. It would help if there were separate ajax functions for json, blob and text, so TypeScript could provide better type safety.
Because right now, the ajax usage is not very type safe:
https://github.com/verdaccio/ui/blob/f84571251912aae71463c44822c84ca70f8b8c40/src/utils/login.ts#L66
The server could return a `string`, in which case there is no code to handle the case that it actually returned a `string` but `json` was expected. 

So in short, without introducing too many type casts/changes to the existing logic, I changed it to `any` to provide a minimum of type safety with the current layout of the code.

Reply:
Thanks @tmkn :) well explained.